### PR TITLE
BLE error screens + new icons + read JSON UART BLE

### PR DESCRIPTION
**Pull request by albertomota-codes**

### CODE REVIEW

File: Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        waveImageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        bodyLabel.translatesAutoresizingMaskIntoConstraints = false
+        actionButton.translatesAutoresizingMaskIntoConstraints = false
+        
+        let safeContainerArea = containerView.safeAreaLayoutGuide
+        
+        // Background Image
+        NSLayoutConstraint.activate([
```
Comment (uBhappy2) [inline Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift:100]: do you prefer to add autolayout constraints in code? either way is fine. Nice :)

File: Ventivader/Ventivader/Utilities/FontsVentivader.swift [hunk 2]
```diff
@@ -10,10 +10,12 @@ import UIKit
 
 class VentivaderFonts {
```
Comment (uBhappy2) [inline Ventivader/Ventivader/Utilities/FontsVentivader.swift:11]: nice!

File: Ventivader/Ventivader/Utilities/LoadingOverlay.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    //var activityIndicator = UIActivityIndicatorView()
+    
+    private lazy var bleLoaderView: UIView = {
+        let containerView = UIView()
+        containerView.backgroundColor = .white
+        
+        let animationView = UIImageView()
+        animationView.loadGif(name: "bleConnectingAnimation")
+        
+        let titleLabel = UILabel()
```
Comment (uBhappy2) [inline Ventivader/Ventivader/Utilities/LoadingOverlay.swift:25]: would be good to later encapulate this code into a private method  for readbility.

File: Ventivader/Ventivader/Utilities/LoadingOverlay.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        titleLabel.updateFont(name: VentivaderFonts.titleFont)
+        titleLabel.textAlignment =  .center
+        
+        containerView.addSubview(animationView)
+        containerView.addSubview(titleLabel)
+        
+        animationView.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Animation View constraints
```
Comment (uBhappy2) [inline Ventivader/Ventivader/Utilities/LoadingOverlay.swift:36]: later we can refactor this autolayout code into a method.

File: Ventivader/Ventivader/ViewControllers/LiveDataChartViewController.swift [hunk 5]
```diff
@@ -34,19 +34,41 @@ class LiveDataChartViewController: UIViewController {
...
+    }
+    
+    func stopDataFlow() {
+        invalidateTimer()
+    }
+    
+    func startDataFlow() {
+        DispatchQueue.main.async { [weak self] in
+            self?.timer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { [weak self] _ in
+                self?.dataSet.append(Double.random(in: 0.0 ..< 20.0))
```
Comment (uBhappy2) [inline Ventivader/Ventivader/ViewControllers/LiveDataChartViewController.swift:53]: willl this dataset eventually be data coming from sensors and rPi? currently it seems to random value beween 0.0 and 20.0

File: Ventivader/Ventivader/ViewModels/ConnectingBLEViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,20 @@
+////
```
Comment (uBhappy2) [inline Ventivader/Ventivader/ViewModels/ConnectingBLEViewModel.swift:1]: can we delete this file? or do you plan to refactor to a view model?

File: Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift (hunk 1 above)
Comment (albertomota-codes) [reply Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift:100]: I think this is how we do the error screens, in other apps ;) (wink).
So, i just keep that approach.

Comment (cskashyap) [reply Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift:100]: Can we simplify this by adding the Error screen into the storyboard or an xib?

Comment (albertomota-codes) [reply Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift]: Ok, i moved this to an xib.

File: Ventivader/Ventivader/Utilities/LoadingOverlay.swift (hunk 4 above)
Comment (albertomota-codes) [reply Ventivader/Ventivader/Utilities/LoadingOverlay.swift]: I did this refactor

File: Ventivader/Ventivader/ViewControllers/LiveDataChartViewController.swift (hunk 5 above)
Comment (albertomota-codes) [reply Ventivader/Ventivader/ViewControllers/LiveDataChartViewController.swift]: This code was unused, i removed it.

File: Ventivader/Ventivader/ViewModels/ConnectingBLEViewModel.swift (hunk 6 above)
Comment (albertomota-codes) [reply Ventivader/Ventivader/ViewModels/ConnectingBLEViewModel.swift]: Deleted this and the VC

**Pull request merged**